MarvelApiClient/src/test/java/com/karumi/MarvelApiClient/CharacterApiClientTest.java

```diff
@@ -0,0 +1,251 @@
+import com.karumi.marvelapiclient.model.MarvelUrl;
+import com.karumi.marvelapiclient.model.StoryResource;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+import org.junit.Test;
+import retrofit.GsonConverterFactory;
+import retrofit.Retrofit;
+
+import static org.junit.Assert.assertEquals;
+
+public class CharacterApiClientTest extends ApiClientTest {
+
+  private static final String ANY_PUBLIC_KEY = "1234";
+  private static final String ANY_PRIVATE_KEY = "abcd";
+  private static final String ANY_URL = "http://fake.marvel.com/";
+  private static final int ANY_OFFSET = 1;
+  private static final int ANY_LIMIT = 10;
+  private static final int INVALID_LIMIT = 0;
+  private static final String ANY_NAME = "Spider-man";
+  private static final String ANY_START_NAME = "Spider";
+  private static final int ANY_COMIC_1 = 1;
+  private static final int ANY_COMIC_2 = 2;
+  private static final int ANY_SERIE = 1;
+  private static final int ANY_STORY = 1;
+  private static final int ANY_EVENT_1 = 1;
+  private static final int ANY_EVENT_2 = 2;
+  private static final String ANY_MODIFIED_SINCE = "2015-01-09T22:10:45-0800";
+  private static final String ORDER_NAME_DESCEDANT_VALUE = "-name";
+  private static final String INVALID_CHARACTER_ID = "";
+  private static final String ANY_NOT_FOUND_ID = "1234";
+  private static final String THREED_MAN_CHARACTER_ID = "123456";
+
+  @Test public void shouldReturnAValidUrlWhenICallToGetAllWithValidOffsetAndLimit()
+      throws Exception {
+    CharacterApiClient characterApiClient = givenCharacterApiClient();
+    enqueueMockResponse(200);
+
+    characterApiClient.getAll(ANY_OFFSET, ANY_LIMIT);
+
+    assertRequestSentToContains("offset=" + ANY_OFFSET, "limit=" + ANY_LIMIT);
+  }
+
+  @Test(expected = IllegalArgumentException.class) public void shouldThrowExceptionWhenLimitIsZero()
+      throws Exception {
+    CharacterApiClient characterApiClient = givenCharacterApiClient();
+    enqueueMockResponse(200);
+
+    characterApiClient.getAll(ANY_OFFSET, INVALID_LIMIT);
+  }
+
+  @Test public void shouldReturnValidUrlWhenICallToGetAllWithBuilder() throws Exception {
+    CharacterApiClient characterApiClient = givenCharacterApiClient();
+    enqueueMockResponse(200);
+
+    CharactersQuery query = CharactersQuery.Builder.create()
+        .withName(ANY_NAME)
+        .withNameStartWith(ANY_START_NAME)
+        .withLimit(ANY_LIMIT)
+        .withOffset(ANY_OFFSET)
+        .withModifiedSince(getAnyDate())
+        .withOrderBy(CharactersQuery.OrderBy.NAME, false)
+        .addComic(ANY_COMIC_1)
+        .addComic(ANY_COMIC_2)
+        .addEvents(getAnyEvents())
+        .addSerie(ANY_SERIE)
+        .addStory(ANY_STORY)
+        .build();
+    characterApiClient.getAll(query);
+
+    assertRequestSentToContains("offset=" + ANY_OFFSET, "limit=" + ANY_LIMIT, "name=" + ANY_NAME,
```

Comment:
nope nope nope, this will fail when ``HashMap`` changes it's internal representation (I bet this test will behave differently between Java 1.6 and 1.7). Either use a TreeMap in ``CharactersQuery`` or make this assert order-agnostic